plugin/pkg/admission/limitranger/admission.go

```diff
@@ -210,76 +210,115 @@ func NewLimitRanger(actions LimitRangerActions) (*LimitRanger, error) {
 	}, nil
 }
 
-// defaultContainerResourceRequirements returns the default requirements for a container
+type defaultResources struct {
+	pod       api.ResourceRequirements
+	container api.ResourceRequirements
+}
+
+// defaultResourceRequirements returns the default requirements for container,
+// if PodLevelResources is enabled, returns the default requirements for pod
 // the requirement.Limits are taken from the LimitRange defaults (if specified)
 // the requirement.Requests are taken from the LimitRange default request (if specified)
-func defaultContainerResourceRequirements(limitRange *corev1.LimitRange) api.ResourceRequirements {
-	requirements := api.ResourceRequirements{}
-	requirements.Requests = api.ResourceList{}
-	requirements.Limits = api.ResourceList{}
+func defaultResourceRequirements(limitRange *corev1.LimitRange, podLevelResourcesEnabled bool) *defaultResources {
```

Comment:
optional: it might be a little cleaner to pass a limit type as a parameter, and then return the default resources for that type. Then, let the caller decide whether pod defaults should be requested. I think this approach scales better, if defaults for another limit type are added.

Reply:
Done, refactored the logic to make it easier to scale and removed feature logic from unnecessary functions.